File: app/src/main/java/com/cliniva/enventory/viewholder/AddProductHolder.java
package com.cliniva.enventory.viewholder;

import android.databinding.ViewDataBinding;
import android.support.annotation.NonNull;

import com.cliniva.enventory.adapter.base.BaseRecyclerClickListener;
import com.cliniva.enventory.adapter.base.BaseRecyclerViewHolder;
import com.cliniva.enventory.databinding.ItemAddProductBinding;
import com.cliniva.enventory.model.AddProduct;

public class AddProductHolder extends BaseRecyclerViewHolder<AddProduct, BaseRecyclerClickListener<AddProduct>> {

    private ItemAddProductBinding binding;

    public AddProductHolder(@NonNull ViewDataBinding itemView) {
        super(itemView);
        this.binding = (ItemAddProductBinding) itemView;
    }

    @Override
    public void onBindView(AddProduct item, int position, BaseRecyclerClickListener<AddProduct> listener) {
        enableItemViewClick(item, listener);
        binding.setAdd(item);
    }
}

File: app/src/main/java/com/amirsons/inventory/recyclerview/viewholder/AddProductHolder.kt
package com.amirsons.inventory.recyclerview.viewholder

import android.view.View

import com.amirsons.inventory.recyclerview.base.BaseRecyclerClickListener
import com.amirsons.inventory.recyclerview.base.BaseRecyclerViewHolder
import com.amirsons.inventory.datamanager.model.AddProduct
import kotlinx.android.synthetic.main.item_transaction_product.view.*

class AddProductHolder(itemView: View) : BaseRecyclerViewHolder<AddProduct, BaseRecyclerClickListener<AddProduct>>(itemView) {

    override fun onBindView(item: AddProduct, position: Int, listener: BaseRecyclerClickListener<AddProduct>?) {
        enableItemViewClick(item, listener)

        itemView.tv_add_product_name.text = item.name
        itemView.tv_add_product_type.text = item.category
    }
}
